x/mongo/driver/driver.go

```diff
@@ -88,11 +88,24 @@ type Compressor interface {
 	CompressWireMessage(src, dst []byte) ([]byte, error)
 }
 
+// ProcessErrorResult represents the result of a ErrorProcessor.ProcessError() call.
+type ProcessErrorResult int
+
+const (
+	// NoChange indicates that the error did not affect the state of the server.
+	NoChange ProcessErrorResult = iota
```

Comment:
Is there a danger of having this as an iota for potential future behavior changes in SDAM? It might be unlikely enough not to worry about now.

If this did return a struct or something, which had methods, maybe that would make it easier for callers. E.g. right now if a caller wants to check if the server was marked unknown, they have to check if the return is either `ServerMarkedUnknown` or `ConnectionPoolCleared`.

I do not have strong feelings about this, and I am fine keeping this an iota. I just wanted to bring this up for consideration. This is also in x so it should not effect general users.

Reply:
Good point. As discussed offline, functions can be defined for any type, so I added a `ProcessErrorResult.ServerChanged() bool` function in my latest commit and some documentation suggesting that it be used rather than manually checking possible values.